src/mp4atom.cpp

```diff
@@ -235,11 +237,16 @@ void MP4Atom::Read()
                      m_File.GetFilename().c_str(), m_type, m_size);
     }
 
-    ReadProperties();
+    // skip parsing of certain atoms
+    if ( g_parseCallback != nullptr && g_parseCallback( ATOMID(m_type) ) )
```

Comment:
So if the client doesn't set a callback (`g_parseCallback` is `nullptr`), this block won't get executed. That seems wrong because previously, this code would all be executed.

Reply:
Hmm, with this implementation, a skipped atom will be left in a "zombie" state, without properties or children. But it will be in the atom "tree" for the file.

Alternatively, after an atom is read it could be omitted from the children list if it's a zombie. 

Both approaches might be breaking assumptions within MP4v2. Like if it expects certain atoms to exist or be valid (non-zombie). In the code, I see that certain child atoms are "required". Anyways, It seems like the second approach is closer to our intentions. I think it's "safe" in the sense that you could manually delete atoms from the MP4 and MP4v2 already needs to be robust enough to handle that.

Reply:
(First approach might also be fine, but it's not immediately clear)